init.lua

```diff
@@ -169,61 +169,63 @@ function ranks.update_nametag(player)
 			prefix = ""
 		end
 
-		player:set_nametag_attributes({
-			text = prefix..name,
-		})
+		if player then
```

Comment:
Unless I've missed something `player` isn't actually defined anywhere anymore. Define it above the opening clause of the if statement, e.g:
```lua
local player = minetest.get_player_by_name(name)
```

Reply:
This is related to the review above.

Reply:
All that is necessary is to define a local player variable, as I have suggested, placing the definition on the line directly above the `if player then` statement, i.e. line 167.

Reply:
Done!

Reply:
Not quite, `player` still needs to be defined on line 169 as a local variable in this scope, defining it anywhere else would either make it global or just add unnecessary load to fetch the player object so early that it might not be necessary.
Definition should look like this:
```lua
local player = minetest.get_player_by_name(name)
```
(we know that `name` will always be a string because of the if check at the opening of the function)